File: Containers_Most_Water/container_mostwater.dart
import 'dart:math';

class Solution {
  int maxArea(List<int> height) {
    int left = 0;
    int right = height.length - 1;
    int area = 0;

    while (left < right) {
      if (height[left] > height[right]) {
        area = max(area, (right - left) * height[right]);
        right--;
      } else {
        area = max(area, (right - left) * height[left]);
        left++;
      }
    }

    return area;
  }
}

File: Containers_Most_Water/container_mostwater.js
/**
 * @param {number[]} height - The height array of water bar
 * @returns {number}  returns the area in integer that contain most water
 */
var maxArea = function (height) {
	let left = 0;
	let right = height.length - 1;
	let area = 0;

	while (left < right) {
		if (height[left] > height[right]) {
			area = Math.max(area, (right - left) * height[right]);
			right--;
		} else {
			area = Math.max(area, (right - left) * height[left]);
			left++;
		}
	}

	return area;
};
